code/game/machinery/navbeacon.dm

```diff
@@ -188,12 +235,33 @@ Transponder Codes:<UL>"}
 			codes.Remove(codekey)
 			codes[newkey] = newval
 
+//Monkstation edit start
+			if(codes)
+				if(codes["patrol"])
+					if(!GLOB.navbeacons["[z]"])
+						GLOB.navbeacons["[z]"] = list()
+					GLOB.navbeacons["[z]"] += src //Register with the patrol list!
+				else
+					GLOB.navbeacons["[z]"] -= src
+				if(codes["delivery"])
+					GLOB.deliverybeacons += src
+					GLOB.deliverybeacontags += location
+				else
+					GLOB.deliverybeacons -= src
+					GLOB.deliverybeacontags -= location
+
 			updateDialog()
 
 		else if(href_list["delete"])
 			var/codekey = href_list["code"]
 			codes.Remove(codekey)
+			if(!codes["patrol"])
+				GLOB.navbeacons["[z]"] -= src
+			if(!codes["delivery"])
+				GLOB.deliverybeacons -= src
+				GLOB.deliverybeacontags -= location
```

Comment:
```suggestion
//Monkestation edit start			
if(!codes["patrol"])
				GLOB.navbeacons["[z]"] -= src
			if(!codes["delivery"])
				GLOB.deliverybeacons -= src
				GLOB.deliverybeacontags -= location
//Monkestation edit end
```
Indentation is messed up on the first if, make sure to fix after committing this